### ISSUE TITLE
Show SQL mode with pretty print SQL statements

### ISSUE THREAD

**Issue by ftrotter** [CareSet/Zermelo#131]
the folks at doctrine have a nice SQL formatter tool: 
https://github.com/doctrine/sql-formatter

And they are slowly working to ensure that some of my requests are met (I stronly prefer forcing the SQL commands and functions to be upper case). 

I would like to have a url for every report that instead of showing the data view shows the SQL view.. Something like:

https://example.com/ZermeloSQL/ReportName/arg1/arg2?something=this&also=that

Which pretty prints the SQL that would have been generated from any of the following: 

https://example.com/Zermelo/ReportName/arg1/arg2?something=this&also=that
https://example.com/ZermeloCard/ReportName/arg1/arg2?something=this&also=that
https://example.com/ZermeloGraph/ReportName/arg1/arg2?something=this&also=that

It just calls the report and gets all of the SQL that would be generated. The system should be smart enough to display an array of SQL when that is what is returned by the original report.

There should also be a configuration parameter at both the report and the configuration level to turn this functionality off, and the tool should ship with this turned off by default in both places. To do otherwise would be a tremendous security risk. 

But for intranet deployments like ours I can see it being always on...

Regular reports should be "aware of" the fact that "showing SQL" is an option when it is turned on (in both the report itself and in the global config) and present a link "show the SQL that made this report". Ah, even the thought sounds glorious.

**Comment by kchapple** [comment]
@ftrotter I have completed the first part of this, adding the route, view and configuration to display the formatted SQL. I have not added buttons to the views yet.

<img width="902" alt="Screen Shot 2020-09-17 at 1 20 27 PM" src="https://user-images.githubusercontent.com/271809/93505459-e2d60780-f8e8-11ea-86d3-8b87bf1d6d10.png">
<img width="855" alt="Screen Shot 2020-09-17 at 1 20 56 PM" src="https://user-images.githubusercontent.com/271809/93505461-e5386180-f8e8-11ea-8234-141d1fe797e5.png">

**Comment by kchapple** [comment]
I have created issue #134 for the work on the views

**Comment by ftrotter** [comment]
The current implementation of this seems entirely dependant on the report value of isSQLPrintEnabled()

It this is "true" the SQL view will work even if the configuration value is set to "false".

**Comment by kchapple** [comment]
Yup, totally dumb. There should be a guard in the controller so it can’t be overridden in the model. Thanks @ftrotter !
